ldai/tracker.py

```diff
@@ -146,6 +146,21 @@ def track_success(self) -> None:
         self._ld_client.track(
             '$ld:ai:generation', self._context, self.__get_track_data(), 1
         )
+        self._ld_client.track(
+            '$ld:ai:generation:success', self._context, self.__get_track_data(), 1
+        )
+
+    def track_error(self) -> None:
+        """
+        Track an unsuccessful AI generation attempt.
+        """
+        self._summary._success = False
+        self._ld_client.track(
+            '$ld:ai:generation', self._context, self.__get_track_data(), 1
+        )
+        self._ld_client.track(
+            '$ld:ai:generation:error', self._context, self.__get_track_data(), 1
+        )
 
     def track_openai_metrics(self, func):
```

Comment:
Do we need an exception handler in here to track failure? Or return an error from trackDurationOf, so it could handle the error and still we could track failure here?

Reply:
Yes, I do think we should. Let's discuss on slack.

Reply:
Okay, updated.